src/Raven.Client/Documents/Operations/CompareExchange/CompareExchangeValueResultParser.cs

```diff
@@ -56,37 +57,47 @@ public static CompareExchangeValue<T> GetSingleValue(BlittableJsonReaderObject i
 
             var type = typeof(T);
 
+            if (raw == null)
+                return new CompareExchangeValue<T>(key, index, default);
+
+            MetadataAsDictionary metadata;
+            if (raw.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject bjro) == false || bjro == null)
+                metadata = new MetadataAsDictionary();
+            else
+                metadata = new MetadataAsDictionary(bjro);
+
             if (type.GetTypeInfo().IsPrimitive || type == typeof(string))
```

Comment:
Here, you are changing how primitive exchange values are held, right?
How does this work with existing values?

Reply:
This is existing code, I just reorganized the method, got the `raw == null` clause out of all if clauses, and added metadata.

This should have any changes on current behavior of primitive exchange values